backend/dataall/core/environment/services/managed_iam_policies.py

```diff
@@ -19,17 +22,54 @@ def __init__(
         self.environmentUri = environmentUri
         self.resource_prefix = resource_prefix
 
+    @abstractmethod
+    def generate_policy_name(self) -> str:
+        """
+        Returns string and needs to be implemented in the ManagedPolicies inherited classes
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def generate_empty_policy(self) -> dict:
+        """
+        Returns dict and needs to be implemented in the ManagedPolicies inherited classes
+        """
+        raise NotImplementedError
+
+
+class PolicyManager(object):
+    def __init__(
+            self,
+            role_name,
+            account,
+            environmentUri,
+            resource_prefix,
+    ):
+        self.role_name = role_name
+        self.account = account
+        self.environmentUri = environmentUri
+        self.resource_prefix = resource_prefix
+        self.ManagedPolicies = ManagedPolicy.__subclasses__()
+        logger.info(f'Found {len(self.ManagedPolicies)} subclasses of ManagedPolicy')
+        self.initializedPolicies = [self._initialize_policy(policy) for policy in self.ManagedPolicies]
```

Comment:
`initializedPolicies` makes me think that some extra processing/initialisation took place yet you only call the ctor. I'd name it `self.policies`

Reply:
I am not sure I am following you, the initalizedPolicies are ManagedPolicies initialized with values